ddtrace/propagation/http.py

```diff
@@ -132,6 +172,32 @@ def _extract(headers):
             headers,
         )
 
+        meta = None
+        tags_value = _extract_header_value(
+            POSSIBLE_HTTP_HEADER_TAGS,
+            headers,
+            default="",
+        )
+        if tags_value:
+            # Do not fail if the tags are malformed
+            try:
+                # We get a Dict[str, str], but need it to be Dict[Union[str, bytes], str] (e.g. _MetaDictType)
+                meta = cast(
+                    Dict[Union[str, bytes], str],
+                    {
+                        k: v
+                        for (k, v) in decode_tagset_string(tags_value).items()
+                        if k in _DatadogMultiHeader._PROPAGATED_TAGS
+                    },
+                )
+            except TagsetDecodeError:
+                log.debug("failed to decode x-datadog-tags: %r", tags_value, exc_info=True)
+            finally:
+                # Datadog propagation should still attempt to decode even if tag
+                # propagation is disabled
```

Comment:
why? could this be a security vulnerability to ddos a webserver?

Reply:
This is by design as the expected behavior is to propagate tags forward even if those tags are not supported yet by a library.

Reply:
Do we have to decode if that is the case? We should be able to just forward the encoded payload along without decoding, right?

Reply:
We do need to decode as we need to report if there are any errors in decoding.